model/src/main/java/org/projectnessie/api/TreeApi.java

```diff
@@ -192,9 +213,48 @@ Branch commitMultipleOperations(
       @Valid
           @NotNull
           @Pattern(regexp = Validation.REF_NAME_REGEX, message = Validation.REF_NAME_MESSAGE)
-          String branchName,
-      @Valid @NotNull @Pattern(regexp = Validation.HASH_REGEX, message = Validation.HASH_MESSAGE)
-          String expectedHash,
+          String branch,
       @Valid @NotNull Operations operations)
       throws NessieNotFoundException, NessieConflictException;
+
+  /**
+   * This operation returns the {@link Content} for a {@link ContentKey} in a named-reference (a
+   * {@link org.projectnessie.model.Branch} or {@link org.projectnessie.model.Tag}).
+   *
+   * <p>If the table-metadata is tracked globally (Iceberg), Nessie returns a {@link Content}
+   * object, that contains the most up-to-date part for the globally tracked part (Iceberg:
+   * table-metadata) plus the per-Nessie-reference/hash specific part (Iceberg: snapshot-ID,
+   * schema-ID, partition-spec-ID, default-sort-order-ID).
+   *
+   * @param key the {@link ContentKey}s to retrieve
+   * @param ref named-reference to retrieve the content for
+   * @return list of {@link GetMultipleContentsResponse.ContentWithKey}s
+   * @throws NessieNotFoundException if {@code ref} or {@code hashOnRef} does not exist
+   */
+  Content getContent(
+      @Valid ContentKey key,
+      @Valid @Pattern(regexp = Validation.REF_NAME_REGEX, message = Validation.REF_NAME_MESSAGE)
+          String ref)
+      throws NessieNotFoundException;
+
+  /**
+   * Similar to {@link #getContent(ContentKey, String)}, but takes multiple {@link ContentKey}s and
+   * returns the {@link Content} for the one or more {@link ContentKey}s in a named-reference (a
+   * {@link org.projectnessie.model.Branch} or {@link org.projectnessie.model.Tag}).
+   *
+   * <p>If the table-metadata is tracked globally (Iceberg), Nessie returns a {@link Content}
+   * object, that contains the most up-to-date part for the globally tracked part (Iceberg:
+   * table-metadata) plus the per-Nessie-reference/hash specific part (Iceberg: snapshot-id,
+   * schema-id, partition-spec-id, default-sort-order-id).
+   *
+   * @param ref named-reference to retrieve the content for
+   * @param request the {@link ContentKey}s to retrieve
+   * @return list of {@link GetMultipleContentsResponse.ContentWithKey}s
+   * @throws NessieNotFoundException if {@code ref} or {@code hashOnRef} does not exist
+   */
+  GetMultipleContentsResponse getMultipleContents(
```

Comment:
Wonder whether we should change the `CommitMeta` for v2 to be more generic, so these attributes:
* commit message
* map of property name to list of property values
* commit ID
* list of parent commit IDs
Pretty much like Git does. With that, there could be for example a list of authors (think: squashing & merging). WDYT?

Reply:
makes sense :+1:

Reply:
#5424 - I'd like to keep this PR limited to v2 API that cover the same functionality that also exists in v1.